tests/unit/koa.tap.js

```diff
@@ -5,28 +5,34 @@
 
 'use strict'
 
-var tap = require('tap')
-var utils = require('@newrelic/test-utilities')
+const tap = require('tap')
+const utils = require('@newrelic/test-utilities')
 
 utils.tap
 
 tap.test('Koa instrumentation', function (t) {
-  var helper = utils.TestAgent.makeInstrumented()
+  const wrapped = ['createContext', 'use', 'emit']
+  const notWrapped = ['handleRequest', 'listen', 'toJSON', 'inspect', 'callback', 'onerror']
+
+  // Save the original methods, to compare with wrapped ones below
+  const origKoa = require('koa')
+  const origMethods = Object.fromEntries(
+    wrapped.concat(notWrapped).map((method) => [method, origKoa.prototype[method]])
+  )
+
+  const helper = utils.TestAgent.makeInstrumented()
   helper.registerInstrumentation({
     moduleName: 'koa',
     type: 'web-framework',
     onRequire: require('../../lib/instrumentation')
   })
-  var Koa = require('koa')
-
-  var wrapped = ['createContext', 'use', 'emit']
-  var notWrapped = ['handleRequest', 'listen', 'toJSON', 'inspect', 'callback', 'onerror']
+  const Koa = require('koa')
 
   wrapped.forEach(function (method) {
```

Comment:
I'm curious about the reversal in these tests. It looks like they'd work as intended, but it's interesting that we're testing each with the opposite approach from before.

Reply:
Ah, see, a wrapped method is one that does not look like the original function. If it's a different function, it's been wrapped.

The old test considered a wrapped method to be one that had the `__NR_original` property.